packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -60,8 +61,9 @@ export interface AsyncDataOptions<
    * Provide a function which returns cached data.
    * A `null` or `undefined` return value will trigger a fetch.
    * Default is `key => nuxt.isHydrating ? nuxt.payload.data[key] : nuxt.static.data[key]` which only caches data when payloadExtraction is enabled.
+   * triggeredBy is a string that indicates in which case the cached data was requested.
    */
-  getCachedData?: (key: string) => DataT
+  getCachedData?: (key: string, triggeredBy?: GetCachedDataTriggeredBy) => DataT
```

Comment:
in honesty I'm not yet sure about this as an API.

What is the situation where passing `force` wouldn't be good enough?

Reply:
In a scenario where you can't set `force` because `watch` is triggered - https://github.com/nuxt/nuxt/issues/24332#issuecomment-1831016309

Reply:
on the force option for refresh, could we just add the force option and basically ignore hasCachedData when forced? And always respect hasCachedData when calling refresh from watch?

Reply:
Will update the PR 👍